sim-administration/sim-batch-management/es2plus/es2plus.go

```diff
@@ -0,0 +1,401 @@
+	ReleaseFlag       bool           `json:"releaseFlag"`
+}
+
+type ES2PlusConfirmOrderResponse struct {
+	Header         ES2PlusResponseHeader  `json:"header"`
+	Iccid          string         `json:"iccid"`
+	Eid            string         `json:"eid,omitempty"`
+    MatchingId     string         `json:"matchingId,omitempty"`
+    SmdpAddress    string         `json:"smdpAddress,omitempty"`
+}
+
+
+//
+//  Generating new ES2Plus clients
+//
+
+
+type Es2PlusClient struct {
+	httpClient *http.Client
+	hostport string
+	requesterId string
+	printPayload bool
+	printHeaders bool
+}
+
+func Client (certFilePath string, keyFilePath string, hostport string, requesterId string) (*Es2PlusClient) {
+    return &Es2PlusClient {
+        httpClient: newHttpClient(certFilePath, keyFilePath),
+        hostport: hostport,
+        requesterId: requesterId,
+        printPayload: false,
+        printHeaders: false,
+    }
+}
+
+func newHttpClient(certFilePath string, keyFilePath string) *http.Client {
+	cert, err := tls.LoadX509KeyPair(
+		certFilePath,
+		keyFilePath)
+	if err != nil {
+		log.Fatalf("server: loadkeys: %s", err)
+	}
+	config := tls.Config{Certificates: []tls.Certificate{cert}, InsecureSkipVerify: true}
+	client := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &config,
+		},
+	}
+	return client
+}
+
+
+///
+/// Generic protocol code
+///
+
+
+//
+// Function used during debugging to print requests before they are
+// sent over the wire.  Very useful, should not be deleted even though it
+// is not used right now.
+//
+func formatRequest(r *http.Request) string {
+	// Create return string
+	var request []string
+	// Add the request string
+	url := fmt.Sprintf("%v %v %v", r.Method, r.URL, r.Proto)
+	request = append(request, url)
+	// Add the host
+	request = append(request, fmt.Sprintf("Host: %v", r.Host))
+	// Loop through headers
+	for name, headers := range r.Header {
+		name = strings.ToLower(name)
+		for _, h := range headers {
+			request = append(request, fmt.Sprintf("%v: %v", name, h))
+		}
+	}
+
+	// If this is a POST, add post data
+	if r.Method == "POST" {
+		r.ParseForm()
+		request = append(request, "\n")
+		request = append(request, r.Form.Encode())
+	}
+	// Return the request as a string
+	return strings.Join(request, "\n")
+}
+
+func newUuid() (string, error) {
+    uuid, err := uuid.NewRandom()
+    if err != nil {
+    		return "", err
+    	}
+    return uuid.URN(), nil
+}
+
+func newEs2plusHeader(client *Es2PlusClient) (*ES2PlusHeader) {
+   functionCallIdentifier, err := newUuid()
+   if err != nil  {
+        panic(err)
+   }
+   return &ES2PlusHeader{FunctionCallIdentifier: functionCallIdentifier, FunctionRequesterIdentifier: client.requesterId}
+}
+
+
+func marshalUnmarshalGenericEs2plusCommand(client *Es2PlusClient, es2plusCommand string,  payload interface{}, result interface{}) error {
+
+    // Serialize payload as json.
+	jsonStrB, err := json.Marshal(payload)
```

Comment:
It is more idiomatic and efficient to use json.Encoder (and Decoder below) than Marshal (and Unmarshal).